### ISSUE TITLE
Questions regarding losses

### ISSUE THREAD

**Issue by boqchen** [filipbasara0/relic#1]
Hi @filipbasara0 

Thanks for the great code!

I have two questions regarding the code.

First, when you calculate the loss you iterate over two lists, one of the views encoded by online_enc and one by target_enc. However, in the loop it seems like you also calculate RELIC loss on the same view but from different encoders. This is not indicated in the original paper. Can you elaborate more on this?

Second, you left a comment "invariance_loss used only for debug". I wonder for the results you reproduced, did you use this invariance loss term or not? In the paper the authors suggest that the generalizability is not guaranteed without this term. 
 
 Thanks in advance!

**Comment by filipbasara0** [comment]
Hi @boqchen and thank you for the questions!

Regarding the first question - I've expanded on the first paper by including global and local (cropped) views (the multi-crop augmentation). This technique is used in the subsequent ReLICv2 paper. The idea is to encode global views with both target and online networks, while local views are encoded only by the online network. 

Coming back to your question - when calculating the loss, we compare target views to all other target (not calculating the loss on online and target preds of the same global view) and all other local views (notice the if condition on [line 99](https://github.com/filipbasara0/relic/blob/eb9288d29bd306284168ac77d60d449e82b861fd/train.py#L99)). Notice that when we revert to the default setting (only 2 global views), we calculate the loss as in the paper (relic_loss(o1, t2) ; relic_loss(o2, t1)).

Regarding the second question - the invariance loss is always used (see [line 51](https://github.com/filipbasara0/relic/blob/eb9288d29bd306284168ac77d60d449e82b861fd/relic/relic.py#L51)), except if you disable the alpha coefficient by setting it to 0. Although already combined with the contrastive loss, the invariance loss is additionally returned only for logging purposes.

I did run experiments with alpha=0.0 and although the training was less stable and resulted in worse representations, it didn't diverge.

Hope this helps!
